srml/democracy/src/lib.rs

```diff
@@ -113,11 +114,24 @@ decl_module! {
 
 		/// Start a referendum.
 		fn start_referendum(proposal: Box<T::Proposal>, vote_threshold: VoteThreshold) -> Result {
-			Self::inject_referendum(
-				<system::Module<T>>::block_number() + Self::voting_period(),
-				*proposal,
-				vote_threshold
-			).map(|_| ())
+			let public_props = Self::public_props();
+			let mut public_props_into_iter = public_props.into_iter();
+
+			// Only allow starting a referendum for a given proposal if the proposal exists.
+			// Alternatively the council may start referendum directly by calling `internal_start_referendum`
+			if let Some((prop_index, _, _)) = public_props_into_iter.find(|ref v| *proposal == v.1) {
+				if let Some((deposit, depositors)) = <DepositOf<T>>::take(prop_index) {
+					// Emit event that proposal has been added to Table of Referenda
+					Self::deposit_event(RawEvent::Tabled(prop_index, deposit, depositors));
+				}
+				Self::inject_referendum(
+					<system::Module<T>>::block_number() + Self::voting_period(),
+					*proposal,
+					vote_threshold
+				).map(|_| ())
+			} else {
+				Err("Cannot start referendum for given proposal if the proposal does not exist, unless called directly by council")?
```

Comment:
You can remove the `?` here.

Reply:
And isn't the error incorrect? When the council calls this method and the proposal does not exist, it will get the same error? The council would need to call `internal_start_referendum`?

Reply:
Yes, I'll have to re-think about this.
I also wanted to add a test called `starting_referendum_after_proposing_a_proposal_should_work`, but I had difficulty with it

Reply:
Gav just highlighted that `start_referendum` may be used by root to start any new referendum even it it doesn't already exist. The existing code comments say that `internal_start_referendum` _"Can be called directly by the council"_.